make install does not install wasm32v1-none target required by deploy

## Description

The `make install` step does not install the `wasm32v1-none` Rust target, but the deploy script (`scripts/deploy.sh`) requires it to compile Soroban smart contracts to WASM. This causes the deploy step to fail with:

```
error[E0463]: can't find crate for `core`
  |
  = note: the `wasm32v1-none` target may not be installed
  = help: consider downloading the target with `rustup target add wasm32v1-none`
```

## Steps to Reproduce

1. Clone the repo on a fresh environment
2. Run `make install`
3. Run `make circuits-build`
4. Run `./scripts/deploy.sh testnet --deployer <identity> --asp-levels 10 --pool-levels 10 --max-deposit 1000000000 --vk-file scripts/testdata/policy_test_vk.json`

Step 4 fails because `stellar contract build` targets `wasm32v1-none` which was never installed.

## Suggested Fix

Add `rustup target add wasm32v1-none` to the `install` target in the Makefile:

```makefile
.PHONY: install
install:
	@echo "Installing frontend dependencies..."
	@npm install --prefix app
	@rustup target add wasm32v1-none
	@command -v trunk >/dev/null 2>&1 || cargo install trunk --locked
	@command -v wasm-pack >/dev/null 2>&1 || cargo install wasm-pack --locked
```

## Workaround

Run `rustup target add wasm32v1-none` manually before the deploy step.

--- a/Makefile
+++ b/Makefile
@@ -34,7 +34,7 @@ circuits-build:
 install:
 	@echo "Installing frontend dependencies..."
 	@npm install --prefix app
-	@rustup target add wasm32v1-none 2>/dev/null || true
+	@rustup target add wasm32v1-none
 	@command -v trunk >/dev/null 2>&1 || cargo install trunk --locked
 	@command -v wasm-pack >/dev/null 2>&1 || cargo install wasm-pack --locked
 

